src/plugins/event-plugins/__tests__/FetchPlugin.test.ts

```diff
@@ -516,6 +516,119 @@ describe('FetchPlugin tests', () => {
         });
     });
 
+    test('when addXRayTraceIdHeader is false then X-Amzn-Trace-Id header is not added to the HTTP request', async () => {
+        // Init
+        const config: PartialHttpPluginConfig = {
+            addXRayTraceIdHeader: false
+        };
+
+        const plugin: FetchPlugin = new FetchPlugin(config);
+        plugin.load(xRayOnContext);
+
+        const init: RequestInit = {};
+
+        // Run
+        await fetch(URL, init);
+        plugin.disable();
+
+        // Assert
+        expect(init.headers).toEqual(undefined);
+    });
+
+    test('when addXRayTraceIdHeader is false then RequestInit is not added to the HTTP request', async () => {
```

Comment:
suggestion: I see where you're going with these "RequestInit" tests, but I don't think they are verifying quite the right properties. Consider refactoring as follows:

1. when header is disabled then header not added (cases: false, regex empty, regex no match)
2. when header is enabled then header added (cases: true, regex match)
3. when RequestInit is supplied, overwrites header
4. when RequestInit is not supplied, creates new RequestInit

You could also use the `mockFetch` methodology to verify (1) and (2) instead of looking at the RequestInit obect.